File: src/main/java/com/transactionService/project/controller/TransactionController.java
package com.transactionService.project.controller;

import com.transactionService.project.model.Transaction;
import com.transactionService.project.service.TransactionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/transactions")
public class TransactionController {
    @Autowired
    private TransactionService transactionService;

    @PostMapping("/create")
    public Transaction create(@RequestBody Transaction transaction){
        return transactionService.save(transaction);
    }

    @GetMapping("/user/{userId}")
    public List<Transaction> getByUserId(@PathVariable Long userId){
        return transactionService.getByUserId(userId);
    }
}

File: transaction-monitoring-ui/controller/transactionController.js
app.controller("TransactionController", function ($scope, TransactionService) {
    $scope.transactions = [];

    TransactionService.getTransactions().then(function (response) {
        $scope.transactions = response.data;
    })
        .catch(function (error) {
            console.log(error.data);
        });
});
